account_payment_order/models/account_move.py

```diff
@@ -39,6 +39,21 @@ def _compute_payment_order_ok(self):
                 payment_mode = move.payment_mode_id
             move.payment_order_ok = payment_mode.payment_order_ok
 
+    def _get_communication(self):
+        """
+        Retrieve the communication string for the payment order
+        """
+        communication = self.ref or self.name or ""
+        if self.is_invoice():
+            if (self.reference_type or "none") != "none":
+                communication = self.ref
+            elif self.is_purchase_document():
+                communication = self.ref or self.payment_reference
+            else:
+                # Force to only put invoice number here
+                communication = self.payment_reference or self.name
```

Comment:
If we force the invoice number, then it should be `self.name` first.

Reply:
I agree with you, this is the same as it is currently (https://github.com/OCA/bank-payment/blob/14.0/account_payment_order/models/account_move_line.py#L101) and changing it will change the expected result so far, do we change it anyway (set `self.name` first) or leave it as it is by removing the comment to avoid confusion?

Reply:
OK, let's remove the comment.